go/pkg/services/logcollector/server.go

```diff
@@ -236,59 +240,104 @@ func (s *LogCollectorServer) StartLog(ctx context.Context, request *protologcoll
 }
 
 // GetLogs returns the log file contents of length size from an offset, for a given run id
-func (s *LogCollectorServer) GetLogs(ctx context.Context, request *protologcollector.GetLogsRequest) (*protologcollector.GetLogsResponse, error) {
+func (s *LogCollectorServer) GetLogs(request *protologcollector.GetLogsRequest, responseStream protologcollector.LogCollector_GetLogsServer) error {
+
+	ctx := responseStream.Context()
+
 	s.Logger.DebugWithCtx(ctx,
 		"Received Get Log request",
 		"runUID", request.RunUID,
 		"size", request.Size,
 		"offset", request.Offset)
 
-	// validate size is not bigger than the buffer size.
-	// if it is, caller will need to call get logs multiple times
-	if int(request.Size) > s.bufferSizeBytes || request.Size < 0 {
-		err := errors.Errorf("Request size is bigger than buffer size %d", s.bufferSizeBytes)
-		return &protologcollector.GetLogsResponse{
-			Success: false,
-			Error:   err.Error(),
-		}, err
-	}
-
 	// get log file path
 	filePath, err := s.getLogFilePath(ctx, request.RunUID)
 	if err != nil {
-		err := errors.Wrapf(err, "Failed to get log file path for run id %s", request.RunUID)
-		return &protologcollector.GetLogsResponse{
-			Success: false,
-			Error:   err.Error(),
-		}, err
+		return errors.Wrapf(err, "Failed to get log file path for run id %s", request.RunUID)
 	}
 
-	// read log from file
-	buffer, err := s.readLogsFromFile(ctx, request.RunUID, filePath, request.Offset, request.Size)
+	if request.Size == 0 {
+		s.Logger.DebugWithCtx(ctx, "Size is 0, returning empty response")
+		if err := responseStream.Send(&protologcollector.GetLogsResponse{
+			Success: true,
+			Logs:    []byte{},
+		}); err != nil {
+			return errors.Wrapf(err, "Failed to send empty logs to stream for run id %s", request.RunUID)
+		}
+		return nil
+	}
+
+	// open log file and calc its size
+	currentLogFileSize, err := common.GetFileSize(filePath)
 	if err != nil {
-		err := errors.Wrapf(err, "Failed to read logs for run id %s", request.RunUID)
-		return &protologcollector.GetLogsResponse{
-			Success: false,
-			Error:   err.Error(),
-		}, err
+		return errors.Wrapf(err, "Failed to get log file size for run id %s", request.RunUID)
+	}
+
+	// if size < 0 - we read only the logs we have for this moment in time starting from offset, so GetLogs will be finite.
+	// otherwise, we read the only the request size from the offset
+	// TODO: when the sdk/UI will support streaming, we can remove this limitation, and stream the logs continuously
```

Comment:
what is "this limitation" ? what exactly need to be removed? is there more logic needs to be taken (implementation details) on client side?

Reply:
Since the clients don't support streaming, the log collector server responds with a finite stream.
"This limitation" = the finite size, set as the current size of the log file. 
If the clients will support a continuous stream, we'll be able to just keep the stream open, and whenever a log is added it will be sent back. Perhaps we can support i there with a flag of some sort.

In this PR, the API handles the stream by constantly appending to a log byte string until the stream is closed, and then returning the string back to the client.